typescript/infra/scripts/funding/update-balance-thresholds.ts

```diff
@@ -0,0 +1,358 @@
+      continue;
+    }
+
+    proposed[thresholdType] = formatBalanceThreshold(
+      burn *
+        balanceThresholdConfigMapping[thresholdType].dailyRelayerBurnMultiplier,
+    );
+  }
+
+  return proposed;
+}
+
+function checkIfReviewNeeded(
+  proposed: Record<BalanceThresholdType, number>,
+  currentThresholds: Record<BalanceThresholdType, number>,
+  overrideForChain: boolean,
+): boolean {
+  // If we have an override, we want to always accept the proposed values
+  if (overrideForChain) return false;
+
+  // if any of the proposed values are less than (LOW_PROPOSED_THRESHOLD_FACTOR)x of the current, we want to review all thresholds for that chain
+  for (const thresholdType of Object.keys(proposed) as BalanceThresholdType[]) {
+    if (
+      currentThresholds[thresholdType] &&
+      proposed[thresholdType] <
+        currentThresholds[thresholdType] * LOW_PROPOSED_THRESHOLD_FACTOR
+    ) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function buildFinalThresholds(
+  proposed: Record<BalanceThresholdType, number>,
+  currentThresholds: Record<BalanceThresholdType, number>,
+  overrideForChain: boolean,
+): Record<BalanceThresholdType, number> {
+  // If we have an override, we want to always accept the proposed values
+  if (overrideForChain) return proposed;
+
+  const finalThresholds = proposed;
+  // only use the proposed values if they are greater than the current
+  for (const thresholdType of Object.keys(proposed) as BalanceThresholdType[]) {
+    if (currentThresholds[thresholdType]) {
+      finalThresholds[thresholdType] = Math.max(
+        currentThresholds[thresholdType],
+        proposed[thresholdType],
+      );
+    }
+  }
+
+  return finalThresholds;
+}
+
+// review and validation functions
+async function handleUserReviews(
+  chain: string,
+  proposedThresholds: Record<BalanceThresholdType, number>,
+  currentThresholds: Record<BalanceThresholdType, number>,
+): Promise<Record<BalanceThresholdType, number>> {
+  rootLogger.info(
+    `\n*** Chain "${chain}": Some proposed thresholds are >50% lower than the current. ***\n`,
+  );
+
+  printChainThresholds(currentThresholds, proposedThresholds);
+
+  const selectedChoice = await select({
+    message: `[${chain}] These thresholds have a >50% reduction. How would you like to handle them?`,
+    choices: [
+      {
+        name: 'Accept ALL Proposed values',
+        value: UserReview.AcceptAllProposed,
+      },
+      { name: 'Keep ALL Current values', value: UserReview.KeepAllCurrent },
+      {
+        name: 'Manual: Decide individually for each threshold',
+        value: UserReview.Manual,
+      },
+    ],
+  });
+
+  switch (selectedChoice) {
+    case UserReview.AcceptAllProposed:
+      return proposedThresholds;
+    case UserReview.KeepAllCurrent:
+      rootLogger.warn(
+        `Consider add ${chain} to the desiredRelayerBalanceOverrides.json file, so that you are not prompted to review the thresholds in the future.`,
```

Comment:
```suggestion
        `Consider adding ${chain} to the desiredRelayerBalanceOverrides.json file, so that you are not prompted to review the thresholds in the future.`,
```
nit